Add cloud_sql_target field in google_data_loss_prevention_discovery_config

patrickmoy opened the pull request:
**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
dlp: added `cloud_sql_target` field to `google_data_loss_prevention_discovery_config` resource
```

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-10625-old..auto-pr-10625) ( 4 files changed, 1090 insertions(+), 1 deletion(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-10625-old..auto-pr-10625) ( 4 files changed, 1090 insertions(+), 1 deletion(-))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-10625-old..auto-pr-10625) ( 1 file changed, 298 insertions(+))

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-10625-old..auto-pr-10625) ( 4 files changed, 1090 insertions(+), 1 deletion(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-10625-old..auto-pr-10625) ( 4 files changed, 1090 insertions(+), 1 deletion(-))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-10625-old..auto-pr-10625) ( 1 file changed, 298 insertions(+))

modular-magician commented:
#### Tests analytics
Total tests: `66`
Passed tests: `64`
Skipped tests: `0`
Affected tests: `2`

<details><summary>Click here to see the affected service packages</summary><blockquote><ul><li>datalossprevention</li></ul></blockquote></details>

#### Action taken
<details> <summary>Found 2 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccDataLossPreventionDiscoveryConfig_Update|TestAccDataLossPreventionDiscoveryConfig_dlpDiscoveryConfigCloudSqlExample </blockquote></details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
#### Tests analytics
Total tests: `66`
Passed tests: `64`
Skipped tests: `0`
Affected tests: `2`

<details><summary>Click here to see the affected service packages</summary><blockquote><ul><li>datalossprevention</li></ul></blockquote></details>

#### Action taken
<details> <summary>Found 2 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccDataLossPreventionDiscoveryConfig_Update|TestAccDataLossPreventionDiscoveryConfig_dlpDiscoveryConfigCloudSqlExample </blockquote></details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{red}{\textsf{Tests failed during RECORDING mode:}}$
`TestAccDataLossPreventionDiscoveryConfig_Update`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10625/artifacts/74a24cff-71fc-4fe4-ba79-ecf36178bc03/build-log/recording_build/TestAccDataLossPreventionDiscoveryConfig_Update_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10625/artifacts/74a24cff-71fc-4fe4-ba79-ecf36178bc03/recording/TestAccDataLossPreventionDiscoveryConfig_Update.log)]
`TestAccDataLossPreventionDiscoveryConfig_dlpDiscoveryConfigCloudSqlExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10625/artifacts/74a24cff-71fc-4fe4-ba79-ecf36178bc03/build-log/recording_build/TestAccDataLossPreventionDiscoveryConfig_dlpDiscoveryConfigCloudSqlExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10625/artifacts/74a24cff-71fc-4fe4-ba79-ecf36178bc03/recording/TestAccDataLossPreventionDiscoveryConfig_dlpDiscoveryConfigCloudSqlExample.log)]


$\textcolor{red}{\textsf{Please fix these to complete your PR.}}$
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10625/artifacts/74a24cff-71fc-4fe4-ba79-ecf36178bc03/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-10625/artifacts/74a24cff-71fc-4fe4-ba79-ecf36178bc03/recording) for each test

modular-magician commented:
$\textcolor{red}{\textsf{Tests failed during RECORDING mode:}}$
`TestAccDataLossPreventionDiscoveryConfig_Update`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10625/artifacts/8cfe1b0a-3835-4918-921e-dda9c9b1afb3/build-log/recording_build/TestAccDataLossPreventionDiscoveryConfig_Update_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10625/artifacts/8cfe1b0a-3835-4918-921e-dda9c9b1afb3/recording/TestAccDataLossPreventionDiscoveryConfig_Update.log)]
`TestAccDataLossPreventionDiscoveryConfig_dlpDiscoveryConfigCloudSqlExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10625/artifacts/8cfe1b0a-3835-4918-921e-dda9c9b1afb3/build-log/recording_build/TestAccDataLossPreventionDiscoveryConfig_dlpDiscoveryConfigCloudSqlExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10625/artifacts/8cfe1b0a-3835-4918-921e-dda9c9b1afb3/recording/TestAccDataLossPreventionDiscoveryConfig_dlpDiscoveryConfigCloudSqlExample.log)]


$\textcolor{red}{\textsf{Please fix these to complete your PR.}}$
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10625/artifacts/8cfe1b0a-3835-4918-921e-dda9c9b1afb3/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-10625/artifacts/8cfe1b0a-3835-4918-921e-dda9c9b1afb3/recording) for each test

patrickmoy commented:
/gcbrun

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-10625-old..auto-pr-10625) ( 4 files changed, 1090 insertions(+), 1 deletion(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-10625-old..auto-pr-10625) ( 4 files changed, 1090 insertions(+), 1 deletion(-))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-10625-old..auto-pr-10625) ( 1 file changed, 298 insertions(+))

modular-magician commented:
#### Tests analytics
Total tests: `66`
Passed tests: `65`
Skipped tests: `0`
Affected tests: `1`

<details><summary>Click here to see the affected service packages</summary><blockquote><ul><li>datalossprevention</li></ul></blockquote></details>

#### Action taken
<details> <summary>Found 1 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccDataLossPreventionDiscoveryConfig_dlpDiscoveryConfigCloudSqlExample </blockquote></details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{green}{\textsf{Tests passed during RECORDING mode:}}$
`TestAccDataLossPreventionDiscoveryConfig_dlpDiscoveryConfigCloudSqlExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10625/artifacts/718c0511-e001-4e5f-a40d-a2b52f4f8eb6/recording/TestAccDataLossPreventionDiscoveryConfig_dlpDiscoveryConfigCloudSqlExample.log)]


$\textcolor{green}{\textsf{No issues found for passed tests after REPLAYING rerun.}}$

---
$\textcolor{green}{\textsf{All tests passed!}}$
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10625/artifacts/718c0511-e001-4e5f-a40d-a2b52f4f8eb6/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-10625/artifacts/718c0511-e001-4e5f-a40d-a2b52f4f8eb6/recording) for each test

slevenick commented on mmv1/third_party/terraform/services/datalossprevention/resource_data_loss_prevention_discovery_config_test.go:
```diff
@@ -584,3 +620,121 @@ resource "google_data_loss_prevention_discovery_config" "basic" {
+            filter {
+                others {}
+            }
+			generation_cadence {
```
Please fix the spacing in this file, may be a tabs vs spaces issue

slevenick commented on mmv1/third_party/terraform/services/datalossprevention/resource_data_loss_prevention_discovery_config_test.go:
```diff
@@ -584,3 +620,121 @@ resource "google_data_loss_prevention_discovery_config" "basic" {
+                    }
+                }
+            }
+			conditions {
```
Spacing

patrickmoy replied:
Done.

slevenick commented on mmv1/third_party/terraform/services/datalossprevention/resource_data_loss_prevention_discovery_config_test.go:
```diff
@@ -584,3 +620,121 @@ resource "google_data_loss_prevention_discovery_config" "basic" {
+resource "google_data_loss_prevention_discovery_config" "basic" {
+	parent = "projects/%{project}/locations/%{location}"
+	location = "%{location}"
+	status = "RUNNING"
```
Spacing

patrickmoy replied:
Done.

slevenick commented on mmv1/third_party/terraform/services/datalossprevention/resource_data_loss_prevention_discovery_config_test.go:
```diff
@@ -584,3 +620,121 @@ resource "google_data_loss_prevention_discovery_config" "basic" {
+        }
+	}
+	targets {
+        cloud_sql_target {
```
Spacing

patrickmoy replied:
Done.

slevenick reviewed (changes requested):
Overall looks good, just some minor notes. Thanks for adding an update test for this!

patrickmoy commented on mmv1/products/dlp/DiscoveryConfig.yaml:
```diff
@@ -325,6 +330,114 @@ properties:
+          properties:
+            - !ruby/object:Api::Type::NestedObject
+              name: filter
+              description: 'Required. The tables the discovery cadence applies to. The first target with a matching filter will be the one to apply to a table.'
```
Done.

patrickmoy commented:
Made indentation changes, added required tag, and squashed fixes into 1 commit (total = 2: 1 base commit to add support, 1 commit with feedback fixes)

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-10625-old..auto-pr-10625) ( 4 files changed, 1090 insertions(+), 1 deletion(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-10625-old..auto-pr-10625) ( 4 files changed, 1090 insertions(+), 1 deletion(-))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-10625-old..auto-pr-10625) ( 1 file changed, 298 insertions(+))

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-10625-old..auto-pr-10625) ( 4 files changed, 1090 insertions(+), 1 deletion(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-10625-old..auto-pr-10625) ( 4 files changed, 1090 insertions(+), 1 deletion(-))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-10625-old..auto-pr-10625) ( 1 file changed, 298 insertions(+))

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-10625-old..auto-pr-10625) ( 4 files changed, 1090 insertions(+), 1 deletion(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-10625-old..auto-pr-10625) ( 4 files changed, 1090 insertions(+), 1 deletion(-))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-10625-old..auto-pr-10625) ( 1 file changed, 298 insertions(+))

modular-magician commented:
#### Tests analytics
Total tests: `66`
Passed tests: `66`
Skipped tests: `0`
Affected tests: `0`

<details><summary>Click here to see the affected service packages</summary><blockquote><ul><li>datalossprevention</li></ul></blockquote></details>

$\textcolor{green}{\textsf{All tests passed!}}$
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10625/artifacts/c92348ed-4a51-4051-afbf-a960df1ba709/build-log/replaying_test.log)

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-10625-old..auto-pr-10625) ( 4 files changed, 1090 insertions(+), 1 deletion(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-10625-old..auto-pr-10625) ( 4 files changed, 1090 insertions(+), 1 deletion(-))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-10625-old..auto-pr-10625) ( 1 file changed, 298 insertions(+))

modular-magician commented:
#### Tests analytics
Total tests: `66`
Passed tests: `66`
Skipped tests: `0`
Affected tests: `0`

<details><summary>Click here to see the affected service packages</summary><blockquote><ul><li>datalossprevention</li></ul></blockquote></details>

$\textcolor{green}{\textsf{All tests passed!}}$
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10625/artifacts/1b0fa980-584b-4154-9c46-93a2d2c9b5c8/build-log/replaying_test.log)

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-10625-old..auto-pr-10625) ( 4 files changed, 1090 insertions(+), 1 deletion(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-10625-old..auto-pr-10625) ( 4 files changed, 1090 insertions(+), 1 deletion(-))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-10625-old..auto-pr-10625) ( 1 file changed, 298 insertions(+))

modular-magician commented:
#### Tests analytics
Total tests: `66`
Passed tests: `66`
Skipped tests: `0`
Affected tests: `0`

<details><summary>Click here to see the affected service packages</summary><blockquote><ul><li>datalossprevention</li></ul></blockquote></details>

$\textcolor{green}{\textsf{All tests passed!}}$
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10625/artifacts/fad11bae-c416-44ab-a5ba-148d30d9cb10/build-log/replaying_test.log)

modular-magician commented:
#### Tests analytics
Total tests: `66`
Passed tests: `66`
Skipped tests: `0`
Affected tests: `0`

<details><summary>Click here to see the affected service packages</summary><blockquote><ul><li>datalossprevention</li></ul></blockquote></details>

$\textcolor{green}{\textsf{All tests passed!}}$
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10625/artifacts/a5048264-6bea-4fbd-b921-495fc7c96cd4/build-log/replaying_test.log)

modular-magician commented:
#### Tests analytics
Total tests: `66`
Passed tests: `66`
Skipped tests: `0`
Affected tests: `0`

<details><summary>Click here to see the affected service packages</summary><blockquote><ul><li>datalossprevention</li></ul></blockquote></details>

$\textcolor{green}{\textsf{All tests passed!}}$
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10625/artifacts/53df9c3a-af11-41da-bdc1-df047059c656/build-log/replaying_test.log)
